core/modules/color/js/color.js

```diff
@@ -5,165 +5,108 @@
-         }
-      }
-    }
 
     // Set up colorScheme selector.
     $('#edit-scheme', form).change(function () {
-      var schemes = settings.color.schemes, colorScheme = this.options[this.selectedIndex].value;
-      if (colorScheme != '' && schemes[colorScheme]) {
+      var schemeName = this.value;
+      if (schemeName !== '' && schemes[schemeName]) {
         // Get colors of active scheme.
-        colors = schemes[colorScheme];
+        var colors = schemes[schemeName];
         for (var fieldName in colors) {
           if (colors.hasOwnProperty(fieldName)) {
-            callback($('#edit-palette-' + fieldName), colors[fieldName], false, true);
+            var input = $('#edit-palette-' + fieldName);
+            if (input.val() && input.val() != colors[fieldName]) {
+              input.val(colors[fieldName]);
+              if (!Backdrop.featureDetect.inputTypeColor()) {
+                input.spectrum({color: colors[fieldName]});
+              }
+            }
           }
         }
-        preview();
+        update_preview();
       }
     });
 
-    /**
-     * Renders the preview.
-     */
-    function preview() {
-      Backdrop.color.callback(context, settings, form, farb, height, width);
-    }
-
-    /**
-     * Shifts a given color, using a reference pair (ref in HSL).
-     *
-     * This algorithm ensures relative ordering on the saturation and luminance
-     * axes is preserved, and performs a simple hue shift.
-     *
-     * It is also symmetrical. If: shift_color(c, a, b) == d, then
-     * shift_color(d, b, a) == c.
-     */
-    function shift_color(given, ref1, ref2) {
-      var d;
-      // Convert to HSL.
-      given = farb.RGBToHSL(farb.unpack(given));
-
-      // Hue: apply delta.
-      given[0] += ref2[0] - ref1[0];
+    $('#palette input').change(function () {
+      var schemeName =  document.getElementById('edit-scheme').value;
+      var key = this.id.substring(13);
 
-      // Saturation: interpolate.
-      if (ref1[1] == 0 || ref2[1] == 0) {
-        given[1] = ref2[1];
-      }
-      else {
-        d = ref1[1] / ref2[1];
-        if (d > 1) {
-          given[1] /= d;
-        }
-        else {
-          given[1] = 1 - (1 - given[1]) * d;
-        }
+      if (schemeName !== '' && this.value !== schemes[schemeName][key]) {
+        resetScheme();
       }
+      update_preview();
+    });
 
-      // Luminance: interpolate.
-      if (ref1[2] == 0 || ref2[2] == 0) {
-        given[2] = ref2[2];
-      }
-      else {
-        d = ref1[2] / ref2[2];
-        if (d > 1) {
-          given[2] /= d;
-        }
-        else {
-          given[2] = 1 - (1 - given[2]) * d;
-        }
-      }
+    // Setup the preview.
+    $('#system-theme-settings').addClass('has-preview').after(settings.color.preview_markup);
 
-      return farb.pack(farb.HSLToRGB(given));
-    }
+    $(document).ready(function () {
+      update_preview();
+    });
```

Comment:
Why is this `$(document).ready()` necessary? `Backdrop.behaviors.*` are already called on `$(document).ready()`.